docs/learn/expert-guides/deploy-multichain-contracts.md

```diff
@@ -0,0 +1,581 @@
+const NICK_FACTORY_ADDRESS = '0x4e59b44847b379578588920ca78fbf26c0b4956c';
+
+// For more information check: https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-16-UniversalFactory.md
+const LSP16_FACTORY_ADDRESS = '0x1600016e23e25D20CA8759338BfB8A8d11563C4e';
+
+const isNickFactoryDeployed = await checkDeployedCode(NICK_FACTORY_ADDRESS);
+const isLSP16FactoryDeployed = await checkDeployedCode(LSP16_FACTORY_ADDRESS);
+
+console.log('Nick Factory exists: ', isNickFactoryDeployed);
+console.log('LSP16UniversalFactory exists: ', isLSP16FactoryDeployed);
+```
+
+### Deployment of Nick Factory
+
+If `Nick Factory` doesn't exist, we'll need to deploy it:
+
+```js title="main.js"
+const signer = new ethers.Wallet('<private-key>', provider);
+
+if (!isNickFactoryDeployed) {
+  const fundingTx = await signer.sendTransaction({
+    to: '0x3fab184622dc19b6109349b94811493bf2a45362',
+    value: ethers.parseEther('0.009'), // This value should be enough
+    // Check gasLimit and gasPrice to estimate exactly the value: https://github.com/Arachnid/deterministic-deployment-proxy
+  });
+
+  await fundingTx.wait();
+
+  // Sending raw transaction specified by the Nick factory
+  const rawTx =
+    '0xf8a58085174876e800830186a08080b853604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf31ba02222222222222222222222222222222222222222222222222222222222222222a02222222222222222222222222222222222222222222222222222222222222222';
+
+  const sentNickTx = await provider.broadcastTransaction(rawTx);
+  await sentNickTx.wait();
+}
+```
+
+### Deployment of LSP16UniversalFactory
+
+If `LSP16UniversalFactory` doesn't exist, we'll need to deploy it:
+
+```js title="main.js"
+if (!isLSP16FactoryDeployed) {
+  const lsp16Tx = await signer.sendTransaction({
+    to: NICK_FACTORY_ADDRESS,
+    data:
+      '0xfaee762dee0012026f5380724e9744bdc5dd26ecd8f584fe9d72a4170d01c049' + // Standardized Salt
+      '60806040523480156100105...', // Standardized Bytecode
+    // Copy the full bytecode from https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-16-UniversalFactory.md#standardized-bytecode
+  });
+
+  await lsp16Tx.wait();
+}
+```
+
+## Step 2: Deploying TargetContract
+
+### Case 1: Contract with Constructor
+
+After deploying the two factory contracts, we can deploy the target contract and precompute its address. We will start by deploying a contract that has a constructor:
+
+```js title="main.js"
+import { abi as LSP16UniversalFactoryABI } from '@lukso/lsp-smart-contracts/artifacts/LSP16UniversalFactory.json';
+import {
+  abi as TargetContractABI,
+  bytecode as targetContractBytecode,
+} from './TargetContract.json';
+
+// signer and provider and LSP16_FACTORY_ADDRESS available from the last step
+
+const lsp16UniversalFactory = new ethers.Contract(
+  LSP16_FACTORY_ADDRESS,
+  LSP16UniversalFactoryABI,
+  signer,
+);
+
+const constructorArgument = 123;
```

Comment:
What is 123? What does it represent? Is it a dummy value? Or is there a real meaning?

I suggest to add a comment to explain this is dummy value or whatever it is.